pgmpy/estimators/TreeAugmentedNaiveBayesSearch.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+import networkx as nx
+
+from pgmpy.estimators import StructureEstimator
+from sklearn.metrics import mutual_info_score
+from pgmpy.base import DAG
+
+
+class TreeAugmentedNaiveBayesSearch(StructureEstimator):
+    def __init__(self, data, class_node, root_node=None, **kwargs):
+        """
+        Search class for learning tree-augmented naive bayes (TAN) graph structure with a given set of variables.
+
+        Parameters
+        ----------
+        data: pandas DataFrame object
+            dataframe object where each column represents one variable.
+
+        class_node: node
+            Dependent variable of the model (i.e. the class label to predict)
+
+        root_node: node (optional)
+            The root node of the tree structure over the independent variables.  If not specified, then
+            an arbitrary independent variable is selected as the root.
+
+        state_names: dict (optional)
+            A dict indicating, for each variable, the discrete set of states (or values)
+            that the variable can take. If unspecified, the observed values in the data set
+            are taken to be the only possible states.
+
+        complete_samples_only: bool (optional, default `True`)
+            Specifies how to deal with missing data, if present. If set to `True` all rows
+            that contain `np.Nan` somewhere are ignored. If `False` then, for each variable,
+            every row where neither the variable nor its parents are `np.NaN` is used.
+            This sets the behavior of the `state_count`-method.
+
+        References
+        ----------
+        Friedman N, Geiger D and Goldszmidt M (1997). Bayesian network classifiers. Machine Learning 29: 131–163
+
+        """
+        self.class_node = class_node
+        self.root_node = root_node
+
+        super(TreeAugmentedNaiveBayesSearch, self).__init__(data, **kwargs)
+
+    def estimate(self):
```

Comment:
It would be nice if you can add the major steps of the algorithm as comments. Something like:
```
Step 1: Argument checks
Step 2: Create a fully connected graph with edge weights as the mutual info score between them.
Step 3: Get the Maximum Spanning tree and orient the edges.
```